controllers/crdmigrator/crd_migrator.go

```diff
@@ -410,9 +410,9 @@ func (r *CRDMigrator) reconcileStorageVersionMigration(ctx context.Context, crd
 		log.V(4).Info("Migrating to new storage version", gvk.Kind, klog.KObj(u))
 		var err error
 		if migrationConfig.UseStatusForStorageVersionMigration {
-			err = r.Client.Status().Patch(ctx, u, client.Apply, client.FieldOwner("crdmigrator"))
+			err = r.Client.Status().Patch(ctx, u, client.Apply, client.FieldOwner("crdmigrator")) //nolint:staticcheck // currently client.Client.Apply() does not support updating Status subresource
```

Comment:
is the only way to update status via a deprecated method?

Reply:
Sure, Added the CR issue link.

Reply:
Sure will watch out.

Reply:
You can rebase: https://github.com/kubernetes-sigs/cluster-api/pull/12740 :)